interfaces/mount/change.go

```diff
@@ -50,11 +50,43 @@ func (c Change) String() string {
 
 // Needed returns true if the change needs to be performed in the context of mount table.
 func (c Change) Needed(mounted []*InfoEntry) bool {
-	// Look through what is mounted and see if we shold perform the change. If
-	// the entry is already mounted then we don't need to mount it, if the
-	// entry is already unmounted then we don't need to unmount it.
+	// Use the following rules to compare mountinfo and fstab entries:
+	// - the mount directory must be the same
+	// - the filesystem type must be the same
+	// - the mounted entity (device) must be the same
+	// - the mounted root must be / itself
+	//
+	// NOTE: This disregards mount options and and any optional fields that
+	// indicate sharing. Snapd doesn't use those properties yet, except for
+	// read-only bind mounts but those are special anyway.
```

Comment:
On second thought I think we do need to look at flags. This is also made a little bit more complicated because there are two sets of flags (superblock and mount) and they mean different things.